### ISSUE TITLE
error: use of undeclared identifier 'size_t'; did you mean 'sizeof'?

### ISSUE THREAD

**Issue by yurivict** [siconos/siconos#410]
```
/disk-samsung/freebsd-ports/science/siconos/work/siconos-4.4.0.rc3/numerics/src/tools/GAMSlink.c:11:3: error: use of undeclared identifier 'size_t'; did you mean 'sizeof'?
  size_t len = strlen(src) + 1;
  ^~~~~~
  sizeof
/disk-samsung/freebsd-ports/science/siconos/work/siconos-4.4.0.rc3/numerics/src/tools/GAMSlink.c:11:10: error: use of undeclared identifier 'len'
  size_t len = strlen(src) + 1;
         ^
/disk-samsung/freebsd-ports/science/siconos/work/siconos-4.4.0.rc3/numerics/src/tools/GAMSlink.c:11:16: warning: implicitly declaring library function 'strlen' with type 'unsigned long (const char *)' [-Wimplicit-function-declaration]
  size_t len = strlen(src) + 1;
               ^
/disk-samsung/freebsd-ports/science/siconos/work/siconos-4.4.0.rc3/numerics/src/tools/GAMSlink.c:11:16: note: include the header <string.h> or explicitly provide a declaration for 'strlen'
/disk-samsung/freebsd-ports/science/siconos/work/siconos-4.4.0.rc3/numerics/src/tools/GAMSlink.c:12:23: warning: implicitly declaring library function 'malloc' with type 'void *(unsigned long)' [-Wimplicit-function-declaration]
  char* dest = (char*)malloc(len * sizeof(char));
                      ^
```

Version 4.4.0.rc3
clang-11
FreeBSD 13

**Comment by xhub** [comment]
Thanks for the report and also porting siconos to FreeBSD!

I have a tentative fix in the CI pipeline that fixes the missing include as well as the logic for detecting the presence of `strdup`.
The libc on FreeBSD does provide this function.

**Comment by xhub** [comment]
The fix rather changes the `#if (__linux ||  __APPLE__)` condition to include the *BSD and also moves the requires includes up in case the libc does not provide `strdup` (which must be pretty rare those days).

Since FreeBSD does provide `strdup` in essence your patch is functionally equivalent to the fix.

**Comment by xhub** [comment]
This should be fixed in 05b48fd. Many thanks for your packaging efforts!

**Comment by yurivict** [comment]
You are welcome!
